### ISSUE TITLE
admin audit of chat histories does not work

### ISSUE THREAD

**Issue by EinsteinGit** [muchino/confluence.chat#110]
As Admin, I go to Configurarion -> Chat -> Chat Histories and choose "Show All Users"

I see a list of users who I know have had chats. So far so good.

Then I choose one of these users who I happen to know that he has never had a chat with admin, but has chatted to other users. I am taken to the user's profile. I then press the "chat history" tab. At this point I expect to see the chat history of the user I chose. Instead what I get is the admin user's own chat history.

It looks like the designer/developer intended the admin to have access to any chosen user's chat history, but instead the admin user get his own chat history listed under the chosen user's profile.

It looks like a bug, rather than a missing feature.

I am using Confluence 5.7 and Confluence Chat 2.06.

**Comment by EinsteinGit** [comment]
I tried building the plugin locally and found the following:

Packaging it with Confluence 4.3, everything works fine, but when installing the plugin to my confluence 5.7.1 I get the above problem with admin audit of user histories.

Packaging it with Confluence 5.7.1, I get build failure with the following complaints:

[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /C:/Documents/Eclipse/NewEclipseWorkspace/confluence.chat/src/main/java/confluence/chat/actions/GetUserHistoryAction.java:[91,17] getUser() in confluence.chat.actions.GetUserHistoryAction cannot override getUser() in com.atlassian.confluence.user.actions.AbstractUserProfileAction
  return type com.atlassian.user.User is not compatible with com.atlassian.confluence.user.ConfluenceUser
[INFO] 1 error

There seems to be something in getting user history which is not compatible with confluence 5.7.1

**Comment by muchino** [comment]
For the tab thing - you are correct I have to change this. 

For the compilation - are you using the SDK or a server instance?

**Comment by EinsteinGit** [comment]
I compiled with the SDK (atlas-mvn) - but I tried the packaged plugin on a server instance.

When I replaced the deprecated getRemoteUser with recommended getAuthenticatedUser, it compiled fine, but the behaviour was the same.

If I run the request:
http://123.123.123.123:8090/users/chat/history.action?username=user2
as admin, I get the admin's chat history. I would have expected user2's history (or be told that I don't have enough privileges).

If I run the request:
http://123.123.123.123:8090/users/chat/history.action?username=user2
(or http://123.123.123.123:8090/users/chat/history.action)
as user2, I get user2's chat history.
